core/metrics-core-service/src/main/java/org/hawkular/metrics/core/jobs/CompressData.java

```diff
@@ -48,18 +50,39 @@
     public static final String JOB_NAME = "COMPRESS_DATA";
     public static final String ENABLED_CONFIG = "compression.enabled";
     public static final String BLOCK_SIZE = "compression.block.size";
+    public static final String CONFIG_ID = "org.hawkular.metrics.jobs." + JOB_NAME;
+
+    private static final int DEFAULT_PAGE_SIZE = 1000;
 
     // TODO Make this configurable by reading BLOCK_SIZE from ConfigurationService
     public static final Duration DEFAULT_BLOCK_SIZE = Duration.standardHours(2);
 
     private MetricsService metricsService;
 
-    public CompressData(MetricsService service) {
+    private int pageSize;
+
+    private boolean enabled;
+
+    public CompressData(MetricsService service, ConfigurationService configurationService) {
         metricsService = service;
+        Configuration configuration = configurationService.load(CONFIG_ID).toSingle().toBlocking().value();
```

Comment:
Shouldn't this be in the call() method? Otherwise it can't be re-enabled without recreating the object.